### DISCUSSION TITLE
Birthday Notification

### DISCUSSION THREAD

**Discussion by decoded-cipher** [decoded-cipher/chacko-mash#24] [General]
Post a notification about the birthdays (today & tomorrow) in the `#inobot` channel, so that no one's birthdays be left unnoticed.
- Post as an embed message with the **Full Name**, **Discord ID** & **Birthday**.
- Check whether the user has a profile pic o not.
  - If `YES`, provide a recommended command so that the admin could copy-paste & execute it _(Embed with GREEN)_.
  - If `NO`, mark it as not recommended to run the command as User Pic is missing _(Embedd with RED)_.

**Comment by decoded-cipher** [comment]
![image](https://user-images.githubusercontent.com/44474792/166591140-9a857243-9adc-447c-80bb-13e45e94630c.png)
![image](https://user-images.githubusercontent.com/44474792/166591173-07e3428b-c4c4-451f-a353-94be3a311f04.png)

**Comment by decoded-cipher** [comment]
Post the birthday notification at a fixed time. For example, ```post the notification at sharp 12:05 AM everyday```, if someone is celebrating the birthday.

The idea to getting notified just because of the restart policy of Heroku is not ideal, as we might migrate to a better reliable Cloud Server in the future.

And, now a days, restart of Heroku Server is usually around 11-12 at night, which is not ideal.

**Comment by Jeevan1310** [comment]
## Need to resolve the glitch in birthday notification 

Cause : Now the bot is only posting birthday notification when we uses the birthday command ( $bday )**

We need the bot to trigger the bday notification exactly at 12:00 am  and must post all possible birthday info in the channel.

**Reply by decoded-cipher** [reply]
- Scheduling isn't a big deal. We could use `cronjob` to schedule the function to be executed at a particular time.
- In general terms, Heroku does have some issues with its free `web-dynos` regarding scheduling, as they fall asleep after some time if left undisturbed. But since our project is a bot - `worker-dyno` it won't sleep at all.

**Reply by decoded-cipher** [reply]
- Error handling in this project _(as of now)_ is not properly given. Nothing's given in `try` - `catch` blocks. Actually, that's the main reason for the glitch that you have experienced.
- And, it also seems to have an issue regarding the `avatar`. That's one main issue up here...

![image](https://user-images.githubusercontent.com/44474792/186511442-8107fd25-ff31-4d0e-9465-1a5fabaedd9b.png)

**Comment by decoded-cipher** [comment]
@Jeevan1310 Issues are fixed.
- The glitch was due to the latency in fetching data via Discord API. An **async-await** is added to resolve the issue _(Hopefully, it should work)_
- A CronJob is added to check for Birthdays' at **00:00** every day.
